experiments/MERA/modified_binary_mera_lib.py

```diff
@@ -0,0 +1,662 @@
+         tf.conj(v_isometry),
+         tf.conj(w_isometry)], indList4)
+
+    return wEnv
+
+
+@tf.contrib.eager.defun
+def get_env_v_isometry(hamAB, hamBA, rhoBA, rhoAB, w_isometry, v_isometry,
+                       unitary):
+
+    indList1 = [[6, 4, 1, 3], [9, 11, 8, -3], [1, 2, 8], [6, 7, 9],
+                [3, 5, 2, -2], [4, 5, 7, 10], [-1, 10, 11]]
+    indList2 = [[3, 4, 1, 2], [8, 10, 9, -3], [5, 6, 9], [5, 7, 8],
+                [1, 2, 6, -2], [3, 4, 7, 11], [-1, 11, 10]]
+    indList3 = [[9, 10, 11, -1], [3, 4, 2, -3], [1, 8, 2], [1, 5, 3],
+                [7, 11, 8, -2], [7, 9, 5, 6], [10, 6, 4]]
+    indList4 = [[7, 5, -1, 4], [6, 3, -3, 2], [7, -2, 6], [4, 1, 2], [5, 1, 3]]
+
+    vEnv = tn.ncon([
+        hamAB, rhoBA, w_isometry,
+        tf.conj(w_isometry), unitary,
+        tf.conj(unitary),
+        tf.conj(v_isometry)
+    ], indList1)
+    vEnv = vEnv + tn.ncon([
+        hamBA, rhoBA, w_isometry,
+        tf.conj(w_isometry), unitary,
+        tf.conj(unitary),
+        tf.conj(v_isometry)
+    ], indList2)
+    vEnv = vEnv + tn.ncon([
+        hamAB, rhoBA, w_isometry,
+        tf.conj(w_isometry), unitary,
+        tf.conj(unitary),
+        tf.conj(v_isometry)
+    ], indList3)
+    vEnv = vEnv + tn.ncon(
+        [hamBA, rhoAB,
+         tf.conj(v_isometry), w_isometry,
+         tf.conj(w_isometry)], indList4)
+
+    return vEnv
+
+
+@tf.contrib.eager.defun
+def steady_state_density_matrices(nsteps, rhoAB, rhoBA, w_isometry, v_isometry,
+                                  unitary, refsym):
+    for n in range(nsteps):
+        rhoAB, rhoBA = descending_super_operator(rhoAB, rhoBA, w_isometry,
+                                                 v_isometry, unitary, refsym)
+        rhoAB = 1 / 2 * (rhoAB + tf.conj(tf.transpose(
+            rhoAB, (2, 3, 0, 1)))) / tn.ncon([rhoAB], [[1, 2, 1, 2]])
+        rhoBA = 1 / 2 * (rhoBA + tf.conj(tf.transpose(
+            rhoBA, (2, 3, 0, 1)))) / tn.ncon([rhoBA], [[1, 2, 1, 2]])
+        if refsym:
+            rhoAB = 0.5 * rhoAB + 0.5 * tf.transpose(rhoAB, (1, 0, 3, 2))
+            rhoBA = 0.5 * rhoBA + 0.5 * tf.transpose(rhoBA, (1, 0, 3, 2))
+    return rhoAB, rhoBA
+
+
+def optimize_mod_binary_mera(hamAB_0,
+                             hamBA_0,
+                             rhoAB_0,
+                             rhoBA_0,
+                             wC,
+                             vC,
+                             uC,
+                             numiter=1000,
+                             refsym=True,
+                             nsteps_steady_state=8,
+                             verbose=0,
+                             opt_u=True,
+                             opt_vw=True,
+                             numpy_update=True,
+                             opt_all_layers=False,
+                             opt_u_after=9):
+    """
+    ------------------------
+    adapted from Glen Evenbly (c) for www.tensors.net, (v1.1) - last modified 24/1/2019
```

Comment:
Try to follow the Google style python docstring. We'll be enforcing it soon on the entire codebase.
https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html